diff --git a/dist/source/Icon.jsx b/dist/source/Icon.jsx
index ec568b63f49d99a09ecef454582fa01fe422c05d..9b35b2a057b410c5a728323ebc31d1d0f04ea3c2 100644
--- a/dist/source/Icon.jsx
+++ b/dist/source/Icon.jsx
@@ -6,8 +6,8 @@
 */
 
 // src/Icon.tsx
-import { createMemo, For, splitProps, useContext } from "solid-js";
-import { Dynamic } from "solid-js/web";
+import { createMemo, For, omit, useContext } from "solid-js";
+import { Dynamic } from "@solidjs/web";
 
 // ../shared/src/build/defaultAttributes.ts
 var defaultAttributes = {
@@ -95,7 +95,9 @@ var hasA11yProp = (props) => {
 // src/Icon.tsx
 import { LucideContext } from "./context";
 var Icon = (props) => {
-  const [localProps, rest] = splitProps(props, [
+  const localProps = props;
+  const rest = omit(
+    props,
     "color",
     "size",
     "width",
@@ -107,7 +109,7 @@ var Icon = (props) => {
     "iconNode",
     "absoluteStrokeWidth",
     "nonScalingStroke"
-  ]);
+  );
   const globalProps = useContext(LucideContext);
   const icon = createMemo(
     () => localProps.icon ?? {
diff --git a/dist/source/context.jsx b/dist/source/context.jsx
index f25d23c1e5a96c282169939f85c320c3483fb8cb..b5a6a7a6c4c905b12526e5699a3500761da78bdb 100644
--- a/dist/source/context.jsx
+++ b/dist/source/context.jsx
@@ -6,7 +6,7 @@
 */
 
 // src/context.tsx
-import { createContext, splitProps } from "solid-js";
+import { createContext, omit } from "solid-js";
 var LucideContext = createContext({
   size: 24,
   color: "currentColor",
@@ -16,15 +16,8 @@ var LucideContext = createContext({
   class: ""
 });
 function LucideProvider(props) {
-  const [value, rest] = splitProps(props, [
-    "size",
-    "color",
-    "strokeWidth",
-    "absoluteStrokeWidth",
-    "nonScalingStroke",
-    "class"
-  ]);
-  return <LucideContext.Provider value={value}>{rest.children}</LucideContext.Provider>;
+  const value = omit(props, "children");
+  return <LucideContext value={value}>{props.children}</LucideContext>;
 }
 export {
   LucideContext,
diff --git a/package.json b/package.json
index 3c8994471fbbf3c27d43f0872a26b8f9a09160d1..0652b2e4c3a2e8861235e899d9513c00734e463c 100644
--- a/package.json
+++ b/package.json
@@ -75,7 +75,8 @@
     "@lucide/shared": "1.0.0"
   },
   "peerDependencies": {
-    "solid-js": "^1.4.7"
+    "@solidjs/web": "^2.0.0-rc.0",
+    "solid-js": "^2.0.0-rc.0"
   },
   "scripts": {
     "build": "pnpm clean && pnpm copy:license && pnpm build:icons && pnpm build:bundle",
